neuralpp/symbolic/sympy_expression.py

```diff
@@ -309,19 +306,13 @@ def from_sympy_function_and_general_arguments(sympy_function: sympy.Basic, funct
         sympy_arguments = [SymPyExpression._convert(argument) for argument in arguments]
         type_dict = build_type_dict_from_sympy_arguments(sympy_arguments)
 
-        # a hack here, because in sympy, "-x" is turned into "(-1)*x", so the function type is actually that
-        # of multiplication. (Similar things happen for "Sub", but since the function type is the same we don't
-        # have to do anything).
-        if sympy_function == sympy_Neg:
-            function_type = Callable[[int, int], int]
```

Comment:
No longer need this because internally it's already `(-1)*x` in sympy, so when we infer the type we get `int->int->int`.